packages/paste-core/components/data-grid/src/table/Th.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import type {BoxElementProps} from '@twilio-paste/box';
+import type {TextAlign} from '@twilio-paste/style-props';
+
+export interface ThProps {
+  textAlign?: TextAlign;
+  width?: string;
+  onMouseDown?: React.MouseEventHandler;
+  element?: BoxElementProps['element'];
+}
+
+export const Th = React.forwardRef<HTMLTableCellElement, ThProps>(
+  ({width, textAlign = 'left', element = 'DATAGRID_TH', ...props}, ref) => {
+    return (
+      <Box
+        {...safelySpreadBoxProps(props)}
+        ref={ref}
+        element={element}
+        as="th"
+        fontSize="fontSize30"
+        lineHeight="lineHeight30"
+        fontWeight="fontWeightSemibold"
+        paddingY="space40"
+        paddingX="space50"
+        width={width}
+        position="relative"
+        textAlign={textAlign}
+        verticalAlign="inherit"
+        _first={{
+          paddingLeft: 'space50',
+        }}
+        _last={{
+          paddingRight: 'space50',
+        }}
+        _focus={{
+          outline: 'none',
+          boxShadow: 'shadowFocusInset',
+        }}
+      />
+    );
+  }
+);
+
+Th.displayName = 'Th';
```

Comment:
This display name, and the TD and TR alike, might clash with the table ones. Lets say for exxample you for what ever reason add a table and a data grid in the same page. Should these be somehting more related to Datagrid so the disaplay names don't clash?

Reply:
It shouldn't be an issue even if they clash, and currently thead/tbody/tfoot/table are just using the Table component so if I change these, the other components will still match Tables. I think it will be fine as is.